chain/semver_test.go

```diff
@@ -0,0 +1,64 @@
+package chain
+
+import (
+	"testing"
+)
+
+func TestSemverCompatible(t *testing.T) {
+	testCases := []struct {
+		name     string
+		required semver
+		actual   semver
+		expected bool
+	}{
+		{
+			name:     "Equal Versions",
+			required: semver{Major: 1, Minor: 2, Patch: 3},
+			actual:   semver{Major: 1, Minor: 2, Patch: 3},
+			expected: true,
+		},
+		{
+			name:     "Different versions",
+			required: semver{Major: 1, Minor: 2, Patch: 3},
+			actual:   semver{Major: 2, Minor: 0, Patch: 0},
+			expected: false,
+		},
```

Comment:
This is the only significant issue I can find while reviewing.

For completeness, you need to add cases where Major is equal and {Minor,Patch} are higher/smaller with the expected result for them.

Reply:
@matheusd i can use do something like these
`required: semver{Major: 1, Minor: 2, Patch: 4},`
`actual:   semver{Major: 1, Minor: 0, Patch: 2},`

`required: semver{Major: 1, Minor: 3, Patch: 6},`
`actual:   semver{Major: 1, Minor: 3, Patch: 2},`

Reply:
Yes, those seem to fill the missing gaps.